hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/AbstractSerializationService.java

```diff
@@ -323,44 +325,44 @@ protected final int calculatePartitionHash(Object obj, PartitioningStrategy stra
     }
 
     protected final boolean safeRegister(final Class type, final Serializer serializer) {
-        return safeRegister(type, createSerializerAdapter(serializer, this));
+        return safeRegister(type, asStreamSerializer(serializer));
     }
 
-    protected final boolean safeRegister(final Class type, final SerializerAdapter serializer) {
+    protected final boolean safeRegister(final Class type, final StreamSerializer serializer) {
         if (constantTypesMap.containsKey(type)) {
             throw new IllegalArgumentException("[" + type + "] serializer cannot be overridden!");
         }
-        SerializerAdapter current = typeMap.putIfAbsent(type, serializer);
-        if (current != null && current.getImpl().getClass() != serializer.getImpl().getClass()) {
+        StreamSerializer current = typeMap.putIfAbsent(type, serializer);
+        if (current != null && current.getClass() != serializer.getClass()) {
```

Comment:
There seems to be a behaviour change. If user configures two different ByteArraySerializer for same type , it will not get an exception. Because both classes will be `ByteArraySerializerStreamSerializer` . I reproduced this in `AbstractSerializationServiceTest` as unittest. But when I try to do a integration test, I could not because there seems to be another check for this in `SerializerHookLoader.register`

```
   @Test(expected = IllegalStateException.class)
    public void testGlobalRegister_doubleRegistration_byteArraySerializer() throws Exception {
        abstractSerializationService.safeRegister(SampleObjects.Employee.class, new FooByteArraySerializer());
        abstractSerializationService.safeRegister(SampleObjects.Employee.class, new BarByteArraySerializer());
    }
```